packages/aws-cdk-lib/aws-logs/lib/log-group.ts

```diff
@@ -236,7 +236,7 @@ abstract class LogGroupBase extends Resource implements ILogGroup {
       return new iam.ArnPrincipal(principal.principalAccount);
     }
 
-    if (principal instanceof iam.ArnPrincipal) {
+    if (principal instanceof iam.ArnPrincipal && principal.arn !== '*') {
```

Comment:
Can you please fix the typo in the function name? `convertArnPrincipalToAccountId`

Reply:
I'll push a change for this in a separate PR. Agreed that it's annoying but I don't want to hold this PR up because of it.